policy-service/README.md

```diff
@@ -13,17 +13,166 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 --->
+
+# <img src="../logos/logo.svg" width="180">
+
 # Policy Service
 
-<span style="color:red">**Note:** As noted in the [documentation root](../README.md) Palisade is in an early stage of development, therefore the precise APIs contained in these service modules are likely to adapt as the project matures.</span>
+The responsibility of the Policy Service is to provide the set of rules (filters or transformations) that need to be
+applied to each resource that has been requested, based the User and Context.
+
+## High Level Architecture
+
+<!--- 
+See policy-service/doc/policy-service.drawio for the source of this diagram
+--->
+![Policy Service diagram](doc/policy-service.png)
+
+## Message Model
+
+| PolicyRequest           | PolicyResponse           | AuditErrorMessage |
+|:------------------------|:-------------------------|:------------------|
+| *token                  | *token                   | *token            |
+| userId                  | userId                   | userId            |
+| resourceId              | resourceId               | resourceId        |
+| context                 | context                  | context           |
+| user                    | user                     | exception         |
+| resource                | resource                 | serverMetadata    |
+|                         | rules                    |                   |
+
+(fields marked with * are acquired from headers metadata)
+
+The service accepts a `PolicyRequest` and a token created by the [Palisade Service](../palisade-serivce), then checks for rules, or policies in the cache; the technology of which is chosen in the Policy Services application.yaml
+and Spring profiles. 
+First, the Policy Service looks for resource level rules, these are rules that can be applied to any part of the resource path, and could limit what resources 
+the User has access to. The Policy Service recursively checks the parents of the resource for any record level rules, and applies them to the resource before continuing.
+If no record or resource level rules are found for this resource, a `NoSuchPolicyException` is thrown.
+An `AuditErrorMessage` is then built, containing the user information, context, resourceId, the error thrown, and any additional attributes. 
+This error object is then sent with the `PolicyRequest` to the [Audit service](../audit-service) to be audited appropriately.
+If however, resource level rules are found, and when applied against the resource, the resource has not been redacted, then the Policy Service will check the cache for Record Level rules. 
+Record Level Rules are rules that can be applied against the data itself, and based off context can limit what the user can see in a particular resource.
```

Comment:
Mention of fine-grained here maybe?
These two (fine and coarse) need only be mentioned as a sort of `..., sometimes referred to as 'fine-grained', ...`

Reply:
added above